### DISCUSSION TITLE
API key not being saved. No key found.

### DISCUSSION THREAD

**Discussion by Leonardo-Alcantar** [danny-avila/LibreChat#4369] [Q&A]
Hello,

I set up a LibreChat instance in Railway. It's running smoothly, and all services are online. However, when I add the API key and submit it, testing the chat results in a “something went wrong. Here’s the specific error message we encountered” message. When I try another test, it says “no key found. Please provide a key and try again.”

I’ve tried providing the key several times by clicking on the OpenAI model and selecting “Set API key.” However, the issue persists. I haven’t changed any environment variables.

![image](https://github.com/user-attachments/assets/d2c2ac64-e5a7-4e0e-97b5-4d0f16fadf19)

**Comment by Iflorisme** [comment]
did you could fix it? same problem here.

**Comment by danny-avila** [comment]
Which expiry option are you using when saving it? I can’t reproduce this issue

**Reply by Leonardo-Alcantar** [reply]
I tried all expiry options. Adding the keys to the variables worked.

**Comment by MintOwlTech** [comment]
Same here. Tried for both OpenAI and Anthropic. Deployed on Railway

**Comment by bdmorin** [comment]
Same. I can't get it to work.

**Comment by EmilianoGarciaLopez** [comment]
Same issue here. Also on Railway

**Reply by EmilianoGarciaLopez** [reply]
Adding the keys as environment variables worked, but I'm running a multi-user system, so I'd like to allow the users to add their own keys

**Reply by georganix** [reply]
Any answers to this?

**Comment by Zvezke** [comment]
Hmm ... adding the OpenAI key changed the error from 'No key found' to 'Something went wrong. Here's the specific error message we encountered: An error occurred while processing your request. Please contact the Admin.'

... has anyone got this working?

I do get Anthropic working by adding the key directly to variables in Railway - not OpenAI, though.

**Comment by Zaldor** [comment]
SAME, anyone has a fix?

**Comment by julianjuenemann** [comment]
same error for me

**Comment by someguy0110** [comment]
Check your MongoDB is connecting.

**Comment by swapnilmmane** [comment]
Hi, I encountered the same error on a fresh installation as well on Railway. 
I updated the `OPENAI_API_KEY` value in the LibraChat variables, redeployed, and this resolved the issue for me.

<img width="1401" height="756" alt="LibreChat Railway" src="https://github.com/user-attachments/assets/22df5554-fb4e-456f-8643-36869255ac90" />

**Reply by sagivo** [reply]
works!
